internal/framework/status/updater_test.go

```diff
@@ -470,6 +470,31 @@ var _ = Describe("Updater", func() {
 				Expect(helpers.Diff(expectedHR, latestHR)).To(BeEmpty())
 			})
 		})
+
+		It("should not update NginxGateway status with canceled context - function normally returns", func() {
+			ctx, cancel := context.WithCancel(context.Background())
+			cancel()
+			updater.Update(ctx, createNGStatus(2))
+		})
+
+		When("updating with canceled context", func() {
+			It("should not have the updated status of the NginxGateway in the API server", func() {
+				latestNG := &ngfAPI.NginxGateway{}
+				expectedNG := createExpectedNGWithGeneration(1)
+
+				err := client.Get(
+					context.Background(),
+					types.NamespacedName{Namespace: "nginx-gateway", Name: "nginx-gateway-config"},
+					latestNG,
+				)
+				Expect(err).Should(Not(HaveOccurred()))
```

Comment:
```suggestion
				Expect(err).ToNot(HaveOccurred())
```

Reply:
I'm seeing a lot of `Expect(err).Should(Not(HaveOccurred()))` in the file and there doesn't seem to be any other cases of using ToNot here, would you still like me to change it?

Reply:
For consistency, I'd say to update those.